pkg/dockerutil/client.go

```
@@ -72,7 +73,7 @@ func (c *Client) PullImage(ctx context.Context, opts docker.PullImageOptions) er
 		reg = "https://index.docker.io/v1/"
 	}
 
-	if c, ok := c.Auth.Configs[reg]; ok {
+	if c, err := c.AuthProvider.GetAuthConfiguration(reg); err == nil {
```

We should bubble this error up if it's non-nil.